ngraph/core/reference/include/ngraph/runtime/reference/einsum.hpp

```diff
@@ -0,0 +1,1210 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <algorithm>
+#include <ngraph/opsets/opset7.hpp>
+
+#include "ngraph/runtime/reference/broadcast.hpp"
+#include "ngraph/runtime/reference/matmul.hpp"
+#include "ngraph/runtime/reference/multiply.hpp"
+#include "ngraph/runtime/reference/reshape.hpp"
+#include "ngraph/runtime/reference/sum.hpp"
+#include "ngraph/runtime/reference/transpose.hpp"
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            namespace einsum_details
+            {
+                std::vector<std::pair<size_t, size_t>> compute_einsum_path(size_t num_inputs)
+                {
+                    // TODO: implement algorithm for finding (pseudo-)optimal einsum_path
+                    std::vector<std::pair<size_t, size_t>> einsum_path;
+                    NGRAPH_CHECK(num_inputs > 0);
+                    for (size_t input_ind = num_inputs - 1; input_ind > 0; --input_ind)
+                    {
+                        einsum_path.push_back(std::make_pair(0, input_ind));
```

Comment:
is it correct that `einsum_path.size()` will be equal `num_inputs - 1`?

Reply:
Exactly correct because the number of operand pairs `num_inputs - 1`